mssql liquibase 4.20.0 - precondition uniqueConstraintExists fails when the unique constraint exists

- [X] I searched and no similar issues were found

Hello! thank you for this api and for your time.
i have a changeset that creates a unique constraint

        <addUniqueConstraint columnNames="NAME" constraintName="UN_TABLENAME_NAME" deferrable="false"
                             disabled="false" initiallyDeferred="false" tableName="TABLENAME"/>
i have a changeset with a precondition on unique constraint exists that fails even though the unique constraint index was created previously

    <changeSet id="CHANGESETID" author="CHANGESETAUTHOR">
        <preConditions>
            <uniqueConstraintExists tableName="TABLENAME" constraintName="UN_TABLENAME_NAME"/>
        </preConditions>
        <dropUniqueConstraint tableName="TABLENAME" constraintName="UN_TABLENAME_NAME"/>
    </changeSet>


- the application does not start because the precondition uniqueConstraintExists is not met

### Steps To Reproduce

the previous changesets will prevent the application from starting with the following stacktrace

```

Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'baseLiquibase' defined in class path resource [...]: Invocation of init method failed; nested exception is liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset changesetfile::CHANGESETID::CHANGESETAUTHOR:
     Reason: 
          changesetfile : UN_TABLENAME_NAME on TABLENAME() does not exist

	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1804)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.addCandidateEntry(DefaultListableBeanFactory.java:1607)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.findAutowireCandidates(DefaultListableBeanFactory.java:1571)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveMultipleBeans(DefaultListableBeanFactory.java:1460)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1347)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1309)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:887)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:791)
	... 101 common frames omitted
Caused by: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset changesetfile::CHANGESETID::CHANGESETAUTHOR:
     Reason: 
          changesetfile : UN_TABLENAME_NAME on TABLENAME() does not exist

	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:136)
	at liquibase.Liquibase.lambda$null$0(Liquibase.java:297)
	at liquibase.Scope.lambda$child$0(Scope.java:187)
	at liquibase.Scope.child(Scope.java:196)
	at liquibase.Scope.child(Scope.java:186)
	at liquibase.Scope.child(Scope.java:165)
	at liquibase.Scope.child(Scope.java:256)
	at liquibase.Liquibase.lambda$update$1(Liquibase.java:295)
	at liquibase.Scope.lambda$child$0(Scope.java:187)
	at liquibase.Scope.child(Scope.java:196)
	at liquibase.Scope.child(Scope.java:186)
	at liquibase.Scope.child(Scope.java:165)
	at liquibase.Liquibase.runInScope(Liquibase.java:2639)
	at liquibase.Liquibase.update(Liquibase.java:236)
	at liquibase.Liquibase.update(Liquibase.java:221)
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:329)
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:284)
	at ....afterPropertiesSet(....java:82)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800)
	... 115 common frames omitted
Caused by: liquibase.exception.MigrationFailedException: Migration failed for changeset changesetfile::CHANGESETID::CHANGESETAUTHOR:
     Reason: 
          changesetfile : UN_TABLENAME_NAME on TABLENAME() does not exist

	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:631)
	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:59)
	at liquibase.changelog.ChangeLogIterator$2.lambda$null$0(ChangeLogIterator.java:124)
	at liquibase.Scope.lambda$child$0(Scope.java:187)
	at liquibase.Scope.child(Scope.java:196)
	at liquibase.Scope.child(Scope.java:186)
	at liquibase.Scope.child(Scope.java:165)
	at liquibase.changelog.ChangeLogIterator$2.lambda$run$1(ChangeLogIterator.java:124)
	at liquibase.Scope.lambda$child$0(Scope.java:187)
	at liquibase.Scope.child(Scope.java:196)
	at liquibase.Scope.child(Scope.java:186)
	at liquibase.Scope.child(Scope.java:165)
	at liquibase.Scope.child(Scope.java:256)
	at liquibase.changelog.ChangeLogIterator$2.run(ChangeLogIterator.java:105)
	at liquibase.Scope.lambda$child$0(Scope.java:187)
	at liquibase.Scope.child(Scope.java:196)
	at liquibase.Scope.child(Scope.java:186)
	at liquibase.Scope.child(Scope.java:165)
	at liquibase.Scope.child(Scope.java:256)
	at liquibase.Scope.child(Scope.java:260)
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:75)
	... 134 common frames omitted
Caused by: liquibase.exception.PreconditionFailedException: Preconditions Failed
	at liquibase.precondition.core.AndPrecondition.check(AndPrecondition.java:47)
	at liquibase.precondition.core.PreconditionContainer.check(PreconditionContainer.java:213)
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:617)
	... 154 common frames omitted


Process finished with exit code 255


```

### Expected/Desired Behavior

we expect that the precondition uniqueConstraintExists is met and the application starts.

### Liquibase Version

4.20.0

### Database Vendor & Version

mssql 19

### Liquibase Integration

spring boot

### Liquibase Extensions

_No response_

### OS and/or Infrastructure Type/Provider

mac m1, mssql 19 docker image

### Additional Context

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR (Thank you!)

--- a/liquibase-integration-tests/src/test/resources/changelogs/common/common.tests.changelog.xml
+++ b/liquibase-integration-tests/src/test/resources/changelogs/common/common.tests.changelog.xml
@@ -1350,4 +1350,10 @@
         </preConditions>
     </changeSet>
 
+    <changeSet id="check-uniqueConstraintExists" author="mpvvliet">
+        <preConditions>
+            <uniqueConstraintExists tableName="createTableNamedUqConst" constraintName="uq_uqtest1"/>
+        </preConditions>
+    </changeSet>
+
 </databaseChangeLog>

--- a/liquibase-standard/src/main/java/liquibase/precondition/core/UniqueConstraintExistsPrecondition.java
+++ b/liquibase-standard/src/main/java/liquibase/precondition/core/UniqueConstraintExistsPrecondition.java
@@ -97,10 +97,15 @@ public class UniqueConstraintExistsPrecondition extends AbstractPrecondition {
 	public void check(Database database, DatabaseChangeLog changeLog, ChangeSet changeSet, ChangeExecListener changeExecListener)
 		throws PreconditionFailedException, PreconditionErrorException {
 
+		String schemaName = getSchemaName();
+		if (schemaName == null) {
+			schemaName = database.getDefaultSchemaName();
+		}
+
 		UniqueConstraint example = new UniqueConstraint(
 			StringUtil.trimToNull(getConstraintName()),
 			StringUtil.trimToNull(getCatalogName()),
-			StringUtil.trimToNull(getSchemaName()),
+			StringUtil.trimToNull(schemaName),
 			StringUtil.trimToNull(getTableName()));
 
 		String columnNames = StringUtil.trimToNull(getColumnNames());

--- a/liquibase-standard/src/main/java/liquibase/precondition/core/UniqueConstraintExistsPrecondition.java
+++ b/liquibase-standard/src/main/java/liquibase/precondition/core/UniqueConstraintExistsPrecondition.java
@@ -97,10 +97,7 @@ public class UniqueConstraintExistsPrecondition extends AbstractPrecondition {
 	public void check(Database database, DatabaseChangeLog changeLog, ChangeSet changeSet, ChangeExecListener changeExecListener)
 		throws PreconditionFailedException, PreconditionErrorException {
 
-		String schemaName = getSchemaName();
-		if (schemaName == null) {
-			schemaName = database.getDefaultSchemaName();
-		}
+		String schemaName = getSchema(database);
 
 		UniqueConstraint example = new UniqueConstraint(
 			StringUtil.trimToNull(getConstraintName()),
@@ -120,6 +117,14 @@ public class UniqueConstraintExistsPrecondition extends AbstractPrecondition {
 		} catch (DatabaseException | InvalidExampleException e) {
 			throw new PreconditionErrorException(e, changeLog, this);
 		}
+	}
+
+	private String getSchema(Database database) {
+		String schemaName = getSchemaName();
+		if (schemaName == null) {
+			schemaName = database.getDefaultSchemaName();
+		}
+		return schemaName;
 	}
 
 	private List<Column> toColumns(Database database, String columnNames) {

--- a/liquibase-standard/src/main/java/liquibase/snapshot/jvm/UniqueConstraintSnapshotGenerator.java
+++ b/liquibase-standard/src/main/java/liquibase/snapshot/jvm/UniqueConstraintSnapshotGenerator.java
@@ -172,7 +172,7 @@ public class UniqueConstraintSnapshotGenerator extends JdbcSnapshotGenerator {
                         + "on const.constraint_schema=col.constraint_schema "
                         + "and const.table_name=col.table_name "
                         + "and const.constraint_name=col.constraint_name "
-                        + "where const.constraint_schema='" + database.correctObjectName(schema.getCatalogName(), Catalog.class) + "' ";
+                        + "where const.constraint_schema='" + database.correctObjectName(schema.getSchema().getName(), Schema.class) + "' ";
                 if (!bulkQuery) {
                     sql += "and const.table_name='" + database.correctObjectName(example.getRelation().getName(), Table.class) + "' "
                             + "and const.constraint_name='" + database.correctObjectName(name, UniqueConstraint.class) + "'";

